File: sdk/android-sdk/src/main/kotlin/com/example/cesiumpoc/cesium_native_android_poc/CesiumGestureOptions.kt
package com.example.cesiumpoc.cesium_native_android_poc

/**
 * V1 gesture configuration surface.
 *
 * Pan, pinch zoom, double-tap zoom, optional inertia, two-finger rotate,
 * and two-finger tilt now map to the Android host gesture layer.
 */
data class CesiumGestureOptions(
    val panEnabled: Boolean = true,
    val zoomEnabled: Boolean = true,
    val rotateEnabled: Boolean = false,
    val tiltEnabled: Boolean = false,
    val inertiaEnabled: Boolean = false,
)

File: sdk/flutter-plugin/lib/src/cesium_gesture_options.dart
class CesiumGestureOptions {
  const CesiumGestureOptions({
    this.panEnabled = true,
    this.zoomEnabled = true,
    this.rotateEnabled = false,
    this.tiltEnabled = false,
    this.inertiaEnabled = false,
  });

  final bool panEnabled;
  final bool zoomEnabled;
  final bool rotateEnabled;
  final bool tiltEnabled;
  final bool inertiaEnabled;

  Map<String, Object?> toMap() => {
    'panEnabled': panEnabled,
    'zoomEnabled': zoomEnabled,
    'rotateEnabled': rotateEnabled,
    'tiltEnabled': tiltEnabled,
    'inertiaEnabled': inertiaEnabled,
  };

  factory CesiumGestureOptions.fromMap(Map<String, Object?> map) {
    return CesiumGestureOptions(
      panEnabled: map['panEnabled'] as bool? ?? true,
      zoomEnabled: map['zoomEnabled'] as bool? ?? true,
      rotateEnabled: map['rotateEnabled'] as bool? ?? false,
      tiltEnabled: map['tiltEnabled'] as bool? ?? false,
      inertiaEnabled: map['inertiaEnabled'] as bool? ?? false,
    );
  }
}
